### PR TITLE
FAL-1496: Introduce XBlockParseException exception

### PR DESCRIPTION
**Pull request by s0b0lev**
Adding an XBlockParseException which would be used to catch xblock parsing failures across several services/projects.

A question to reviewers:
- Should I bump the package version within this commit? 

**Reviewers:**
- @eLRuLL 
- edX reviewer

### CODE REVIEW

**Comment by edx-webhook** [comment]
Thanks for the pull request, @s0b0lev! I've created [OSPR-5680](https://openedx.atlassian.net/browse/OSPR-5680) to keep track of it in JIRA, where we prioritize reviews. Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).


Please let us know once your PR is ready for our review and all tests are green.

**Comment by natabene** [comment]
@s0b0lev Thank you for your contribution. Please let me know once it is ready for our review.

**Comment by eLRuLL** [comment]
@s0b0lev yes I was asked on other PRs to bump the version with the PR as well, please also update the CHANGELOG file.

**Comment by s0b0lev** [comment]
Thanks, @eLRuLL. The package version was bumped and CHANGELOG updated. 

If everything is ok, could you put approval on further review by edX?

**Comment by eLRuLL** [comment]
@s0b0lev yep LGTM :+1: but I think the branch is not updated? @natabene this should be ready for edX review

**Comment by natabene** [comment]
@eLRuLL Thanks, I will line this up for our review.

**Comment by pomegranited** [comment]
@natabene  @symbolist has asked if he can review this as core committer. Is that ok?

**Comment by natabene** [comment]
@s0b0lev Let me check and get back to you on this.

**Comment by natabene** [comment]
@s0b0lev Yep, that is fine.

**Comment by symbolist** [comment]
@ormsbee This is coming up to address this issue that happened a few weeks ago: https://openedx.slack.com/archives/C01ATGRBBDM/p1614180754002200

The other related PRs are:
* https://github.com/edx/edx-platform/pull/27068
* https://github.com/open-craft/labxchange-xblocks/pull/17

File: xblock/exceptions.py
Comment (symbolist) [inline xblock/exceptions.py:143]: Can you make it `Raised if parsing the XBlock olx fails.`?

File: CHANGELOG.rst
Comment (symbolist) [inline CHANGELOG.rst:10]: Nit: missing fullstop.

**Review by symbolist — CHANGES_REQUESTED** [review]

File: xblock/exceptions.py
Comment (s0b0lev) [reply xblock/exceptions.py]: Updated :ok_hand:

File: CHANGELOG.rst
Comment (s0b0lev) [reply CHANGELOG.rst]: Updated :ok_hand:

**Review by symbolist — APPROVED** [review]
@s0b0lev Thanks! Looks like there is a merge conflict. Can you rebase and (remove the JIRA id from the commit message)? I'll merge once it is green.

**Comment by edx-webhook** [comment]
@s0b0lev 🎉 Your pull request was merged!

Please take a moment to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=edx/XBlock&entry.867055334=https://github.com/edx/XBlock/pull/516&entry.1484655318=https://github.com/s0b0lev&entry.752974735=2021-03-18+19:48&entry.1917517419=2021-04-05+10:50&entry.2133058324=Yes) so we can improve your experience in the future.

**Comment by s0b0lev** [comment]
@symbolist can we deploy update to pypi?

**Comment by symbolist** [comment]
@s0b0lev Yup, needs push rights to the repo (https://github.com/edx/XBlock#packaging). I have deployed v1.4.1: https://pypi.org/project/XBlock/

**Pull request merged**
